packages/firestore/src/lite-api/geo_point.ts

```diff
@@ -94,4 +89,53 @@ export class GeoPoint {
       primitiveComparator(this._long, other._long)
     );
   }
+
+  /** Returns a JSON-serializable representation of this GeoPoint. */
+  toJSON(): { latitude: number; longitude: number; type: string } {
+    return {
+      latitude: this._lat,
+      longitude: this._long,
+      type: 'firestore/geopoint/1.0'
+    };
+  }
+
+  /** Builds a `Timestamp` instance from a JSON serialized version of `Bytes`. */
+  static fromJSON(json: object): GeoPoint {
+    const requiredFields = ['type', 'latitude', 'longitude'];
+    let error: string | undefined = undefined;
+    let lat: number = 0;
+    let long: number = 0;
+    for (const key of requiredFields) {
+      if (!(key in json)) {
+        error = `json missing required field: ${key}`;
+      }
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const value = (json as any)[key];
+      if (key === 'type') {
+        if (typeof value !== 'string') {
+          error = `json field 'type' must be a string.`;
+          break;
+        } else if (value !== 'firestore/geopoint/1.0') {
```

Comment:
I'm wondering if we should have this json have a version number independent of the SDK version.

If they are independent, that means JSON can be deserialized and reserialized across SDK versions... Maybe that's okay.

But if they are the same, that means the same SDK must be used for serialization and deserialization, which would also free us up to make changes more easily

Reply:
We spoke about this offline.